src/ex4/dist/main.js

```diff
@@ -1,52 +1,111 @@
-class Main {
+import ItemClient from "./clients/item_client.js";
+
+// Implement the `Main` class here
+class Main{
+
     constructor() {
-        this.itemClient = new ItemClient()
+        this.itemClient = new ItemClient();
     }
 
-    init = async () => {
-        const addItemButton = document.getElementById("list-item-submit");
-        addItemButton.addEventListener("click", this.handleItem);
+    toggleLoader(state){
+        document.getElementById("todoList").hidden = state;
+        document.querySelector(".loader").hidden = !state;
+    }
+    toggleTodoList(state){
+        document.getElementById("todoList").hidden = !state;
+    }
+    async renderTodoList() {
+        this.toggleLoader(true);
+        this.toggleTodoList(false);
+        const todoList = await this.itemClient.getTodos();
```

Comment:
Why do we need to get the todos from the server for every action we're doing?
Try to minimize calls to server when needed